test/cpp/callbackcontext.cpp

```diff
@@ -0,0 +1,66 @@
+/*********************************************************************
+ * NAN - Native Abstractions for Node.js
+ *
+ * Copyright (c) 2018 NAN contributors
+ *
+ * MIT License <https://github.com/nodejs/nan/blob/master/LICENSE.md>
+ ********************************************************************/
+
+#include <nan.h>
+#include <unistd.h>
```

Comment:
Needs special-casing for windows.  We have 5 repetitions of that in test/cpp already so it's about time it gets factored out into a header file.

Reply:
Yes, it should be factored out. Windows CI runs are here https://ci.appveyor.com/project/RodVagg/nan

Reply:
I've refactored `Sleep` into `sleep.h`. More aggressive refactoring of `asyncresource.cpp` and `callbackcontext.cpp` is possible, but IMO it would be keep those as is to serve as examples.